File: src/web/magmac/app/lang/CommonRules.ts
import { FilterRule } from "../../../magmac/app/compile/rule/FilterRule";
import { Rule } from "../../../magmac/app/compile/rule/Rule";
import { StringRule } from "../../../magmac/app/compile/rule/StringRule";
import { StripRule } from "../../../magmac/app/compile/rule/StripRule";
import { TypeRule } from "../../../magmac/app/compile/rule/TypeRule";
export class CommonRules {
	createSymbolRule(key : String) : Rule {return new StripRule( FilterRule.Symbol( new StringRule( key)));;}
	createSymbolRule() : Rule {return new TypeRule( "symbol", createSymbolRule( "value"));;}
}

File: src/java/magmac/app/lang/CommonRules.java
package magmac.app.lang;

import magmac.app.compile.rule.FilterRule;
import magmac.app.compile.rule.Rule;
import magmac.app.compile.rule.StringRule;
import magmac.app.compile.rule.StripRule;
import magmac.app.compile.rule.TypeRule;

public class CommonRules {
    public static Rule createSymbolRule(String key) {
        return new StripRule(FilterRule.Symbol(new StringRule(key)));
    }

    public static Rule createSymbolRule() {
        return new TypeRule("symbol", createSymbolRule("value"));
    }
}
